tests/h/util/group_scope_test.py

```diff
@@ -28,6 +28,50 @@ def test_it_parses_scope_from_uri(uri, expected_scope):
     assert scope == expected_scope
 
 
+class TestURIInScope(object):
+    @pytest.mark.parametrize(
+        "uri,in_origin,in_path,in_other",
+        [
+            ("https://www.foo.com", True, False, False),
+            ("http://foo.com", False, False, False),
+            ("http://www.foo.com/bar/qux.html", True, True, True),
+            ("http://www.foo.com/bar/baz/qux.html", True, True, True),
+            ("http://foo.com/", False, False, False),
+            ("http://www.foo.com/bar.baz", True, False, False),
+            ("www.foo.com", False, False, False),
+            ("randoscheme://foo.com", False, False, False),
+            ("foo", False, False, False),
+            ("https://www.foo.com/bar/baz", True, False, False),
+            ("", False, False, False),
+        ],
+    )
+    def test_it_returns_True_if_uri_matches_one_or_more_scopes(
+        self, scope_lists, uri, in_origin, in_path, in_other
+    ):
+        assert scope_util.uri_in_scope(uri, scope_lists["origin_only"]) == in_origin
+        assert scope_util.uri_in_scope(uri, scope_lists["with_path"]) == in_path
+        assert scope_util.uri_in_scope(uri, scope_lists["with_other"]) == in_other
+
+    @pytest.fixture
+    def match(self, patch):
+        return patch("h.util.group_scope.re.match")
+
+    @pytest.fixture
+    def scope_lists(self):
+        return {
+            "origin_only": ["http://www.foo.com", "https://www.foo.com"],
+            "with_path": [
+                "http://www.foo.com/bar/baz",
+                "http://www.foo.com/bar/qux",
+                "http://www.foo.com/bar/baz/qux",
+            ],
+            "with_other": [
+                "http://www.foo.com/bar/baz/qux.html",
+                "http://www.foo.com/bar/qux",
+            ],
```

Comment:
What is the "other" here?  The URLs in "with_other" only have origins and paths.